grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/RedisTaskStorage.java

```diff
@@ -68,18 +72,29 @@ public static RedisTaskStorage create(Pool<Jedis> jedisPool, MetricRegistry metr
 
     @Override
     public TaskId newState(TaskState state) throws GraknBackendException {
-        updateState(state);
-        return state.getId();
+        try(Jedis jedis = redis.getResource(); Context ignore = updateTimer.time()){
+            String key = encodeKey.apply(state.getId().getValue());
+            LOG.debug("New state {}", key);
+            String value = new String(Base64.getEncoder().encode(SerializationUtils.serialize(state)),
+                    Charsets.UTF_8);
+            String status = jedis.set(key, value, "nx", "ex", 60*60/*expire time in seconds*/);
+            if (status != null && status.equalsIgnoreCase("OK")) {
+                return state.getId();
+            } else {
+                LOG.error("Could not write state {} to redis. Returned: {}", key, status);
+                throw new GraknBackendException("Could not save task state " + key);
```

Comment:
Tiny poke again. We usually try to use a factory approach for creating and throwing errors . We find this makes it easier to prevent making duplicate error messages as well as keeping error messages of the same format.